PublicFolders/src/SourceSideValidations/Get-FolderData.ps1

```diff
@@ -21,66 +30,99 @@ function Get-FolderData {
-                    $folderData.NonIpmSubtree | Export-Csv $PSScriptRoot\NonIpmSubtree.csv
-                }
+            if ($null -ne $job.NonIpmSubtree) {
+                $folderData.NonIpmSubtree = $job.NonIpmSubtree
+            }
 
-                if ($null -ne $job.ItemCounts) {
-                    $folderData.ItemCounts = $job.ItemCounts
-                    $folderData.ItemCounts | Export-Csv $PSScriptRoot\ItemCounts.csv
-                }
+            if ($null -ne $job.ItemCounts) {
+                $folderData.ItemCounts = $job.ItemCounts
             }
         }
 
+        $folderData.IpmSubtree | Export-Csv $PSScriptRoot\IpmSubtree.csv
+        $folderData.NonIpmSubtree | Export-Csv $PSScriptRoot\NonIpmSubtree.csv
+
         $folderData.IpmSubtreeByMailbox = $folderData.IpmSubtree | Group-Object ContentMailbox
         $folderData.IpmSubtree | ForEach-Object { $folderData.ParentEntryIdCounts[$_.ParentEntryId] += 1 }
         $folderData.IpmSubtree | ForEach-Object { $folderData.EntryIdDictionary[$_.EntryId] = $_ }
+        # We can't count on $folder.Path.Depth being available in remote powershell,
+        # so we calculate the depth by walking the parent entry IDs.
+        $folderData.IpmSubtree | ForEach-Object {
+            $pathDepth = 0
+            $parent = $folderData.EntryIdDictionary[$_.ParentEntryId]
+            while ($null -ne $parent) {
+                $pathDepth++
+                $parent = $folderData.EntryIdDictionary[$parent.ParentEntryId]
+            }
+
+            Add-Member -InputObject $_ -MemberType NoteProperty -Name FolderPathDepth -Value $pathDepth
+        }
         $folderData.NonIpmSubtree | ForEach-Object { $folderData.NonIpmEntryIdDictionary[$_.EntryId] = $_ }
-        $folderData.ItemCounts | ForEach-Object {
-            if ($_.ItemCount -gt 0) {
-                $folder = $folderData.EntryIdDictionary[$_.EntryId.ToString()]
 
-                if ($null -ne $folder) {
-                    $folder.ItemCount = $_.ItemCount
+        # If we're doing slow traversal, we have to get the stats after we have the hierarchy
+        # grouped by mailbox.
+        if ($SlowTraversal) {
+            if (-not $StartFresh -and (Test-Path $PSScriptRoot\ItemCounts.csv)) {
+                $folderData.ItemCounts = Import-Csv $PSScriptRoot\ItemCounts.csv
+            } else {
+                Write-Verbose "Starting slow traversal item count."
+                $itemCountResult = Get-ItemCount $serverName $folderData
+                $folderData.ItemCounts = $itemCountResult.ItemCounts
+                foreach ($errorParam in $itemCountResult.Errors) {
+                    $errorResult = New-TestResult @errorParam
+                    $folderData.Errors.Add($errorResult)
                 }
             }
         }
+
+        $folderData.ItemCounts | ForEach-Object { $folderData.ItemCountDictionary[$_.EntryId] = $_.ItemCount }
+        $folderData.ItemCounts | Export-Csv $PSScriptRoot\ItemCounts.csv
```

Comment:
Same thing here, we are importing and exporting the same data.

Reply:
Fixed.